### PR TITLE
Issue2

### PR DESCRIPTION
**Pull request by filipkozlik**
-added front and back laser sensor on top of pioneer robot
-position of sensors can be adjusted in /models/ariac/model.sdf
-sensor angle range can be adjusted in /models/custom_hokuyo/model-1_2.sdf

### CODE REVIEW

File: models/ariac/model.sdf [hunk 1]
```diff
@@ -1873,5 +1873,23 @@
     </link>
     <static>1</static>
     <allow_auto_disable>1</allow_auto_disable>
+    <include>
+      <uri>model://custom_hokuyo</uri>
```
Comment (dmiklic) [inline models/ariac/model.sdf]: Seems like you have added the lasers directly into the world model, which seems unnatural. I suggest you create a new xacro model to describe the robot, which will include the pioneer xacro and the hokuyo xacro models.

File: models/custom_hokuyo/model-1_4.sdf [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" ?>
```
Comment (dmiklic) [inline models/custom_hokuyo/model-1_4.sdf]: Why do we have three different model files for the hokuyo, which seem to define very similar stuff?

**Review by dmiklic — CHANGES_REQUESTED** [review]

File: models/ariac/model.sdf (hunk 1 above)
Comment (filipkozlik) [reply models/ariac/model.sdf]: I will change it to the way you specified.

File: models/custom_hokuyo/model-1_4.sdf (hunk 2 above)
Comment (filipkozlik) [reply models/custom_hokuyo/model-1_4.sdf]: Those files were defined by hokuyo themselves to support different sdf versions. I will try to remove unnecessary files.

Comment (dmiklic) [reply models/custom_hokuyo/model-1_4.sdf]: If this is the way the model is defined, it is not necessary that we change it.

File: models/ariac/model.sdf (hunk 1 above)
Comment (dmiklic) [reply models/ariac/model.sdf]: Cool, thanks!

File: launch/p3dx_in_ariac_world.launch [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0"?>
 <launch>
-  <include file="$(find p2os_urdf)/launch/upload_pioneer3dx.xml"/>
+  <include file="$(find ariac_environment)/models/pioneer_sensor_model/launch/upload_pioneer3dx.xml"/>
```
Comment (dmiklic) [inline launch/p3dx_in_ariac_world.launch]: All launch files should go into the top-level launch folder `ariac_environment/launch`

File: models/hokuyo/hokuyo.xacro [hunk 4]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0"?>
```
Comment (dmiklic) [inline models/hokuyo/hokuyo.xacro:1]: Please clean up indentation in the whole file. tabs should be 2 spaces.

File: models/pioneer_sensor_model/defs/pioneer3dx.xacro [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+
+<robot name="pioneer3dx" xmlns:xacro="http://ros.org/wiki/xacro">
+	   
+	<xacro:include filename="$(find ariac_environment)/models/pioneer_sensor_model/defs/pioneer3dx_body.xacro"/>
```
Comment (dmiklic) [inline models/pioneer_sensor_model/defs/pioneer3dx.xacro]: If `pioneer3dx_body.xacro` is identical to the one from the `p2os_urdf` package, we should just include that one.

File: models/pioneer_sensor_model/defs/pioneer3dx.xacro [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+<!-- Front Laser -->
+	<xacro:property name="laser_name" value="front_laser"/>
```
Comment (dmiklic) [inline models/pioneer_sensor_model/defs/pioneer3dx.xacro]: Pls fix indentation.

File: models/pioneer_sensor_model/defs/pioneer3dx.xacro [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+	<xacro:property name="xyz" value="0.1 0 0.3"/>
+	<xacro:property name="rpy" value="0 0 0"/>
+	<xacro:property name="publish_topic" value="/pio_lasers/front_laser"/>
```
Comment (dmiklic) [inline models/pioneer_sensor_model/defs/pioneer3dx.xacro]: Value should be `"scan_front"`. Remove leading `/` it indicates an absolute path and paths should always be defined as relative to enable correct namespace scoping.

File: models/pioneer_sensor_model/defs/pioneer3dx.xacro [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+	<xacro:include filename="$(find ariac_environment)/models/hokuyo/hokuyo.xacro"/>
+
+<!-- Back Laser -->
+	<xacro:property name="laser_name" value="back_laser"/>
+	<xacro:property name="xyz" value="-0.2 0 0.3"/>
+	<xacro:property name="rpy" value="0 0 3.14"/>
+	<xacro:property name="publish_topic" value="/pio_lasers/back_laser"/>
```
Comment (dmiklic) [inline models/pioneer_sensor_model/defs/pioneer3dx.xacro]: Value should be `"scan_back"`, same rationale as above.

File: models/pioneer_sensor_model/defs/pioneer3dx_body.xacro [hunk 9]
```diff
@@ -0,0 +1,264 @@
+<?xml version="1.0"?>
```
Comment (dmiklic) [inline models/pioneer_sensor_model/defs/pioneer3dx_body.xacro]: if this file is identical to the one in `p2os_urdf`, we should delete it and use that one.

File: models/pioneer_sensor_model/launch/upload_pioneer3dx.xml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<launch>
```
Comment (dmiklic) [inline models/pioneer_sensor_model/launch/upload_pioneer3dx.xml]: Move launch file to top-level launch folder.

File: models/pioneer_sensor_model/defs/pioneer3dx.xacro [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
```
Comment (dmiklic) [inline models/pioneer_sensor_model/defs/pioneer3dx.xacro]: Rename folder to `p3dx_with_sensors`

**Review by dmiklic — CHANGES_REQUESTED** [review]

File: models/pioneer_sensor_model/defs/pioneer3dx.xacro (hunk 7 above)
Comment (filipkozlik) [reply models/pioneer_sensor_model/defs/pioneer3dx.xacro]: Should it just be "scan_front" or should I add something in front of it (e.g. "pioneer/scan_front")?

Comment (dmiklic) [reply models/pioneer_sensor_model/defs/pioneer3dx.xacro]: Just scan_front. The prefix can be added at runtime by namespacing or topic renaming.

**Pull request merged**